python/ray/rllib/agent.py

```diff
@@ -38,7 +53,7 @@ class Agent(Trainable):
         registry (obj): Object registry.
     """
 
-    _allow_unknown_configs = False
+    _allow_unknown_configs = True
```

Comment:
why True by default? This catches a lot of user errors

Reply:
The problem this is trying to address is passing in other parameters to `ConfigProto`. Should we just allow users to pass in arbitrary configurations for nested dicts? In that case, we can just close this PR.

Reply:
So I thought this PR was to address the issue of updating one field of a nested pr, e.g. say the default was:

```
{
   foo: {
      bar: 3
      baz: 4
    }
}
```

I should be able to override with `{foo: {bar: 100}}`. That doesn't require setting allow unknown to false right?

Reply:
From the high level, that's right. But take the following example:

```
default = {
   tensorflow_config: {
      bar: 3
      baz: 4
    }
}
```
Right now on `master`, we can do the following:
```
custom = {
   tensorflow_config: {
      bar: 3,
      bar2: 4
      bar3: 5
    }
}
```
By setting unknown to false, we cannot do that any more (with this PR). It doesn't make sense to say that `bar2` and `bar3` are not valid, since tensorflow objects have so many options.